frontend/src/lib/api.ts

```diff
@@ -1,50 +1,6 @@
-// API service for activities
-export type Activity = {
-  id: number;
-  created_at: string;
-  brand_id: string;
-  name: string;
-  start_time: string;
-  end_time: string | null;
-  execution_time_ms: number | null;
-  recording_count: number;
-  has_recording: boolean;
-};
+import type { paths } from "@generated/api";
+import createFetchClient from "openapi-fetch";
+import createClient from "openapi-react-query";
 
-export type RecordingResponse = {
-  events: RRWebEvent[];
-};
-
-export type RRWebEvent = {
-  type: number;
-  data: Record<string, unknown>;
-  timestamp: number;
-  [key: string]: unknown;
-};
-
-export class ApiService {
-  private static baseUrl = "/api";
-
-  static async getActivities(): Promise<Activity[]> {
-    const response = await fetch(`${this.baseUrl}/activities/`);
-
-    if (!response.ok) {
-      throw new Error(`Failed to fetch activities: ${response.statusText}`);
-    }
-
-    const data: Activity[] = await response.json();
-    return data;
-  }
-
-  static async getRecording(activityId: string): Promise<RecordingResponse> {
-    const response = await fetch(
-      `${this.baseUrl}/activities/${activityId}/recordings`,
-    );
-
-    if (!response.ok) {
-      throw new Error(`Failed to fetch recording: ${response.statusText}`);
-    }
-
-    return await response.json();
-  }
-}
+const fetchClient = createFetchClient<paths>({ baseUrl: "/api" });
+export const $api = createClient(fetchClient);
```

Comment:
NIT: can you please remove `$` here?

Reply:
I think this one is specifically what the the lib suggests. Let’s just leave it like this then.